### PR TITLE
block repository testing

### PR DESCRIPTION
**Pull request by alex-arugon**

### CODE REVIEW

File: core/repository/block/pg.ts [hunk 1]
```diff
@@ -133,7 +135,8 @@ class BlockPGRepo implements IBlockPGRepository {
         return rawBlockIds.map(block => block.id);
     }
 
-    async getMany(limit: number, offset: number): Promise<Array<Block>> {
+    async getMany(offset: number = 0, limit: number = 0): Promise<Array<Block>> {
```
Comment (axelsheva) [inline core/repository/block/pg.ts]: Why did you switch parameters?

File: core/repository/block/pg.ts [hunk 2]
```diff
@@ -145,7 +148,7 @@ class BlockPGRepo implements IBlockPGRepository {
 
     async isExist(blockId: BlockId): Promise<boolean> {
         let exists = await db.oneOrNone(queries.isExist, { blockId });
-        return exists === 't';
+        return exists.exists;
```
Comment (axelsheva) [inline core/repository/block/pg.ts]: exists.exists?

File: core/repository/queries/block.ts [hunk 3]
```diff
@@ -12,5 +18,5 @@ export default {
             'ORDER BY "height"'
         ].filter(Boolean).join(' ');
     },
-    isExist: 'SELECT EXISTS(SELECT * FROM block WHERE "id" = ${id})'
+    isExist: 'SELECT EXISTS(SELECT * FROM block WHERE "id" = ${blockId})'
```
Comment (axelsheva) [inline core/repository/queries/block.ts]: '*' is not optimized

File: core/repository/block/pg.ts (hunk 1 above)
Comment (alex-arugon) [reply core/repository/block/pg.ts]: To suit memory repository, and imho better sequence

Comment (Verbalist) [reply core/repository/block/pg.ts]: Better will be general. In general u say limit offset. Without limit request not possible, but without offset possible. Right order `limit, offset`. I fixed it in previous merge. Return to right order please.

File: core/loader.ts [hunk 4]
```diff
@@ -105,7 +105,7 @@ class Loader {
     private async blockWarmUp(limit) {
         let offset: number = 0;
         do {
-            const blockBatch: Array<Block> = await BlockPGRepository.getMany(limit, offset);
+            const blockBatch: Array<Block> = await BlockPGRepository.getMany(offset, limit);
```
Comment (Verbalist) [inline core/loader.ts]: order

File: core/repository/block/index.ts [hunk 5]
```diff
@@ -38,13 +38,15 @@ class BlockRepo implements IBlockRepository {
         return lastBlock;
     }
 
-    public getMany(startHeight: number, limit?: number): Array<Block> {
+    public getMany(startHeight?: number, limit?: number): Array<Block> {
```
Comment (Verbalist) [inline core/repository/block/index.ts]: startHeight -> offset, limit -> required

File: core/repository/block/pg.ts [hunk 6]
```diff
@@ -82,13 +82,15 @@ class BlockPGRepo implements IBlockPGRepository {
...
+    async deleteById(blockId: BlockId | Array<BlockId>): Promise<Array<string>> {
         const blockIds: Array<BlockId> = [].concat(blockId);
-        await db.none(queries.deleteByIds, [blockIds]);
+        const ids = await db.manyOrNone(queries.deleteByIds, [blockIds]);
+        return ids.map(rawId => rawId.id);
     }
 
-    async deleteAfterBlock(blockId: BlockId): Promise<void> {
-        await db.none(queries.deleteAfterBlock, blockId);
+    async deleteAfterBlock(blockId: BlockId): Promise<Array<string>> {
```
Comment (Verbalist) [inline core/repository/block/pg.ts]: deleteAfterBlock it`s redundant logic. Remove it

File: core/repository/block/pg.ts (hunk 2 above)
Comment (Verbalist) [reply core/repository/block/pg.ts]: null.exist can possible

File: test/core/repository/block/block.memory.test.ts [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+                expect(blocks).to.be.lengthOf(1);
+            });
+
+            after(() => {
+                BlockRepo.deleteLastBlock();
+            });
+        });
+    });
+
+    // describe('lastNBlocks usage', () => {
```
Comment (Verbalist) [inline test/core/repository/block/block.memory.test.ts]: remove

File: api/service/block.ts [hunk 8]
```diff
@@ -11,7 +11,7 @@ interface IBlockService {
 class BlockService implements IBlockService {
 
     getMany(data?: reqGetBlocks): ResponseEntity<Array<Block>> {
```
Comment (Verbalist) [inline api/service/block.ts]: in all place limit, offset there data ...

File: core/repository/block/pg.ts [hunk 9]
```diff
@@ -144,8 +140,8 @@ class BlockPGRepo implements IBlockPGRepository {
     }
 
     async isExist(blockId: BlockId): Promise<boolean> {
-        let exists = await db.oneOrNone(queries.isExist, { blockId });
-        return exists === 't';
+        let response = await db.one(queries.isExist, { blockId });
```
Comment (axelsheva) [inline core/repository/block/pg.ts]: Method `one` will be created exceptions when a block will be not exists

File: core/repository/queries/block.ts [hunk 10]
```diff
@@ -1,16 +1,15 @@
...
+    getById: 'SELECT * FROM block WHERE "id" = ${blockId}',
     getGenesisBlock: 'SELECT * FROM block WHERE "height" = 1',
     getLastBlock: 'SELECT * FROM block WHERE "height" = ( SELECT MAX("height") FROM block ) ORDER BY "height"',
     getLastNBlocks: 'SELECT id FROM block ORDER BY height DESC LIMIT ${blockLimit}',
     getMany(limit: number) {
         return [
             'SELECT * FROM block WHERE "height" > ${offset}',
             (limit ? 'AND "height" <= ${offset} + ${limit}' : ''),
-            'ORDER BY "height" LIMIT ${limit}'
+            (limit ? 'ORDER BY "height" LIMIT ${limit}' : '')
```
Comment (axelsheva) [inline core/repository/queries/block.ts]: I think `ORDER BY` should be always and LIMIT need to remove

File: core/repository/block/pg.ts (hunk 9 above)
Comment (axelsheva) [reply core/repository/block/pg.ts]: But I don't know how works `SELECT EXISTS`. Maybe exception will not happen

Comment (alex-arugon) [reply core/repository/block/pg.ts]: This function always returns value

**Pull request merged**